ecommerce/views_test.py

```diff
@@ -0,0 +1,77 @@
+"""
+Tests for ecommerce views
+"""
+from mock import (
+    MagicMock,
+    patch,
+)
+
+from django.core.urlresolvers import reverse
+from django.test import override_settings
+from rest_framework.status import (
+    HTTP_200_OK,
+    HTTP_400_BAD_REQUEST,
+    HTTP_403_FORBIDDEN,
+)
+
+from profiles.factories import UserFactory
+from search.base import ESTestCase
+
+
+CYBERSOURCE_SECURE_ACCEPTANCE_URL = 'http://fake'
+
+
+class CheckoutViewTests(ESTestCase):
+    """
+    Tests for /api/v0/checkout/
+    """
+
+    def test_authenticated(self):
+        """
+        Unauthenticated users can't use this API
+        """
+        resp = self.client.post(reverse('checkout'), {}, format='json')
+        assert resp.status_code == HTTP_403_FORBIDDEN
+
+    def test_valid_course_id(self):
+        """
+        If course_id is not present in payload a ValidationError is raised
+        """
+        user = UserFactory.create()
+        self.client.force_login(user)
+        resp = self.client.post(reverse('checkout'), {}, format='json')
+        assert resp.status_code == HTTP_400_BAD_REQUEST
+        assert resp.json() == ['Missing course_id']
```

Comment:
we should probably implement our own mixin to handle these errors, but not for this PR